microservices/download/utils.js

```diff
@@ -1,4 +1,18 @@
 const https = require("node:https");
+let TOKEN = process.env.GH_TOKEN_DOWNLOAD_MICROSERVICE;
+
+// Environment Variables Check
+
+if (typeof TOKEN === "undefined") {
+  if (process.env.PULSAR_STATUS === "dev") {
+    // We are in dev mode, assign dev values
+    TOKEN = "123456";
+  } else {
+    // We are not in dev mode. Our secrets are gone and the application will fail to work
+    console.log("Missing Required Environment Variables! Something has gone wrong!");
```

Comment:
We're only handling one missing env var for now, and maybe we should handle each one specifically if there are ever more env vars, for the sake of clear and more-actionable logging. (Hopefully something that's obvious  to read and understand what's going on if at 3 AM or tired, etc. etc. etc.)

Point being, maybe this could stand to be a bit more specific. e.g. "Missing GitHub Auth token! Our requests will be un-authenticated and possibly subjected to rate limits for unauthenticated GitHub API access."

And optionally just warn and continue un-authed... If we're comfortable doing that instead of bringing the service down so we notice it's un-authed faster. A genuine trade-off I'm not sure which way side to come down on. But food for thought, maybe.

Depending on how you feel about it, I can PR a change for this, or you're free to do so yourself if that is in fact less hassle than reviewing a PR. Whatever works!

Thanks for addressing this stuff!

Reply:
This isn't a bad idea to make it much more exact.

Although, I'm pretty sure if we add an authorization header when we hit GitHub, if that header is invalid, such as missing, the request will be denied no matter what. Since it's seen as failing auth, rather than a malformed request. So I'd still opt for exiting the whole process if not present. Otherwise I'll address the specificity of error messages in a follow up PR